lib/lambda_ethereum_consensus/state_transition/cache.ex

```diff
@@ -30,9 +62,15 @@ defmodule LambdaEthereumConsensus.StateTransition.Cache do
         value
 
       [] ->
+        clean_up_old_entries(table, key)
         value = compute_fun.()
         :ets.insert_new(table, {key, value})
         value
     end
   end
+
+  defp clean_up_old_entries(table, key) do
+    match_spec = generate_cleanup_spec(table, key)
```

Comment:
is this potentially an expensive operation?

Reply:
It shouldn't be. It is necessary, however, and I already have some thoughts on how to optimize it in case it starts impacting performance.